src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -225,6 +239,14 @@ internal void ValidateOptions(DiagnosticBag diagnostics, CommonMessageProvider m
                 diagnostics.Add(messageProvider.CreateDiagnostic(messageProvider.ERR_InvalidDebugInformationFormat, Location.None, (int)DebugInformationFormat));
             }
 
+            foreach (var instrumentationKind in InstrumentationKinds)
+            {
+                if (!instrumentationKind.IsValid())
+                {
+                    diagnostics.Add(messageProvider.CreateDiagnostic(messageProvider.ERR_InvalidInstrumentationKind, Location.None, (int)instrumentationKind));
```

Comment:
I think this is ok, since the user needs to cast int to the enum type to produce invalid value. All valid values are named (have a enum member). Invalid values are unnamed (don't have enum value), so there is nothing better to display.